src/entities/token.ts

```diff
@@ -56,7 +56,7 @@ export class Token extends BaseCurrency {
    */
   public sortsBefore(other: Token): boolean {
     invariant(this.chainId === other.chainId, 'CHAIN_IDS')
-    invariant(this.address !== other.address, 'ADDRESSES')
+    invariant(this.address.toLowerCase() !== other.address.toLowerCase(), 'ADDRESSES')
     return this.address.toLowerCase() < other.address.toLowerCase()
   }
```

Comment:
might be worth adding a test case for this

Reply:
so true bestie